internal/util/fileop_test.go

```diff
@@ -358,6 +424,154 @@ func TestFileOpCopyBase(test *testing.T) {
 	}
 }
 
+func TestFileOpCopyGlob(test *testing.T) {
+	testCases := []struct {
+		source         string
+		dest           string
+		expectedExists []string
+		errorSubstring string
+	}{
+		{
+			alreadyExistsDirname + "/*",
+			startingEmptyDirname,
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join(startingEmptyDirname, alreadyExistsFilename),
+				filepath.Join(startingEmptyDirname, alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			startingEmptyDirname,
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join(startingEmptyDirname, alreadyExistsFilename),
+				filepath.Join(startingEmptyDirname, alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		// Note, the entire directory is copied into startingEmptyDirname.
+		{
+			alreadyExistsDirname,
+			startingEmptyDirname,
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join(startingEmptyDirname, alreadyExistsFileRelpath),
+				filepath.Join(startingEmptyDirname, alreadyExistsFileAltRelpath),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*",
+			alreadyExistsDirname,
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*",
+			"a",
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join("a", alreadyExistsFilename),
+				filepath.Join("a", alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*",
+			"a.txt",
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join("a.txt", alreadyExistsFilename),
+				filepath.Join("a.txt", alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			"a",
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join("a", alreadyExistsFilename),
+				filepath.Join("a", alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			"a.txt",
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join("a.txt", alreadyExistsFilename),
+				filepath.Join("a.txt", alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			alreadyExistsDirname + "/*.txt",
+			alreadyExistsDirname + "/*.txt",
+			[]string{
+				alreadyExistsFileRelpath,
+				alreadyExistsFileAltRelpath,
+				filepath.Join(alreadyExistsDirname, "*.txt", alreadyExistsFilename),
+				filepath.Join(alreadyExistsDirname, "*.txt", alreadyExistsFilenameAlt),
+			},
+			"",
+		},
+		{
+			"*",
```

Comment:
Add a similar case to this.
You have `copy * a`,
add `copy * <alreadyExistsDirname>`.
This will specifically test ignoring a self copy when doing globs.

Make a move version of this as well.